### PR TITLE
refactor!: Using non const InternalSpacePoint objects

### PR DESCRIPTION
**Pull request by noemina**
This PR tries to fix usage of mutable properties in https://github.com/acts-project/acts/pull/1168 and https://github.com/acts-project/acts/pull/1143.

Tagging: @paulgessinger @robertlangenberg @LuisFelipeCoelho 

BREAKING CHANGE: Remove assumption on constness of InternalSpacePoint. 
The need of this major change is to allow seed finding and filtering to evaluate and set properties of InternalSpacePoints as soon as they are candidates for triplet formation and seed quality evaluation.

### CODE REVIEW

**Review by paulgessinger — COMMENTED** [review]
Looks good overall. I added a few comments.

I believe this breaks backward compatibility, and should be noted. Once we're ready to merge, can you write 2 sentences describing the change, that we can put into the release notes once this goes in?

**Comment by krasznaa** [comment]
@beomki-yeo, @konradkusiak97, could you help Noemi update the CUDA and SYCL seedfinder code in this PR? Unfortunately I think the changes will have to be made in this single PR, as she is changing some interfaces that all implementations of the seed-finding are using. (So we can't easily just patch the CUDA and SYCL codes first, before this change would be made.)

As far as I can see the changes necessary for the CUDA and SYCL plugins should be mostly trivial. (Re)moving some `const` statements here and there. But I thought you two should help out with that. :wink: (By either telling Noemi what she should do in those plugins to make them work with her updates, or by sending her patch files with the changes that you would propose.)

File: Core/include/Acts/Seeding/InternalSpacePoint.hpp [hunk 1]
```diff
@@ -67,6 +76,10 @@ inline InternalSpacePoint<SpacePoint>::InternalSpacePoint(
   m_r = std::sqrt(m_x * m_x + m_y * m_y);
   m_varianceR = variance.x();
   m_varianceZ = variance.y();
+  // initialised to invalid values, will be changed
+  // during seed finding and filtering if needed
+  m_quality = -std::numeric_limits<double>::infinity();
+  m_cotTheta = std::numeric_limits<double>::quiet_NaN();
```
[the remark this answers is not in the document]
Comment (noemina) [reply Core/include/Acts/Seeding/InternalSpacePoint.hpp]: done

File: Core/include/Acts/Seeding/InternalSpacePoint.hpp [hunk 2]
```diff
@@ -40,6 +40,13 @@ class InternalSpacePoint {
   float phi() const { return atan2f(m_y, m_x); }
   const float& varianceR() const { return m_varianceR; }
   const float& varianceZ() const { return m_varianceZ; }
+  const float& quality() const { return m_quality; }
+  const float& cotTheta() const { return m_cotTheta; }
+  void setCotTheta(float cotTheta) { m_cotTheta = cotTheta; }
+  void setQuality(float quality) {
+    if (quality >= m_quality)
+      m_quality = quality;
```
[the remark this answers is not in the document]
Comment (noemina) [reply Core/include/Acts/Seeding/InternalSpacePoint.hpp]: done

File: Core/include/Acts/Seeding/InternalSpacePoint.hpp [hunk 3]
```diff
@@ -83,6 +96,8 @@ inline InternalSpacePoint<SpacePoint>::InternalSpacePoint(
   m_r = sp.m_r;
   m_varianceR = sp.m_varianceR;
   m_varianceZ = sp.m_varianceZ;
+  m_quality = sp.m_quality;
+  m_cotTheta = sp.m_cotTheta;
```
[the remark this answers is not in the document]
Comment (noemina) [reply Core/include/Acts/Seeding/InternalSpacePoint.hpp]: done

**Comment by beomki-yeo** [comment]
Did you change all CUDA 1&2 plugins?

**Comment by konradkusiak97** [comment]
> Did you change all CUDA 1&2 plugins?

Yes

**Comment by beomki-yeo** [comment]
Great Thanks!

**Comment by noemina** [comment]
I have committed the changes from @konradkusiak97. Thank you!

**Review by paulgessinger — APPROVED** [review]
Let's do this then.

**Pull request merged**